crates/egui/src/widgets/drag_value.rs

```diff
@@ -508,8 +508,32 @@ impl<'a> Widget for DragValue<'a> {
 }
 
 fn clamp_to_range(x: f64, range: RangeInclusive<f64>) -> f64 {
-    x.clamp(
-        range.start().min(*range.end()),
-        range.start().max(*range.end()),
-    )
+    match x.total_cmp(range.start()) {
+        Ordering::Less | Ordering::Equal => *range.start(),
+        Ordering::Greater => match x.total_cmp(range.end()) {
+            Ordering::Greater | Ordering::Equal => *range.end(),
+            Ordering::Less => x,
+        },
+    }
```

Comment:
Would this simpler solution also work:

``` rust
let min = range.start().min(*range.end());
let max = range.start().max(*range.end());
if x <= min {
    min
} else if x <= max {
    x
} else {
    max
}
```

I think that should also cover the `-0.0` case

Reply:
Documentation states that `PartialEq` and `PartialOrd` implementation of `f32` and `f64` differs from ordering of [total_cmp](https://doc.rust-lang.org/nightly/std/primitive.f64.html#method.total_cmp). So I believe it won't be as simple as just using operators. 

> The ordering established by this function does not always agree with the [PartialOrd](https://doc.rust-lang.org/nightly/std/cmp/trait.PartialOrd.html) and [PartialEq](https://doc.rust-lang.org/nightly/std/cmp/trait.PartialEq.html) implementations of f64. For example, they consider negative and positive zero equal, while total_cmp doesn’t.

Reply:
Yeah, but I'm pretty sure I set up the code above to work correctly for the `-0` vs `+0` cases (and chose to ignore NaN, because whatever we do there is bad)

Reply:
Yeah you're right, but it won't work in the case of `clamp_to_range(0.0, (-1.0)..=(-0.0))`, although I am not sure if there is any use-case for this kind of input.

Reply:
ah, right. perhaps your original code was better. Best to cover it all with tests though.

Reply:
I think I covered all cases in the new commit